easy-random-core/src/main/java/org/jeasy/random/util/ReflectionUtils.java

```diff
@@ -140,21 +131,16 @@ public Object invoke(final Object proxy, final Method method, final Object[] arg
      * @throws IllegalAccessException if the property cannot be set
      */
     public static void setProperty(final Object object, final Field field, final Object value) throws IllegalAccessException, InvocationTargetException {
-        try { // to call regular setter
-            PropertyDescriptor propertyDescriptor = new PropertyDescriptor(field.getName(), object.getClass());
-            Method setter = propertyDescriptor.getWriteMethod();
+        try {
+            Method setter = getSetter(object.getClass(), field.getName(), field.getType());
             if (setter != null) {
                 setter.invoke(object, value);
             } else {
                 setFieldValue(object, field, value);
             }
-        } catch (IntrospectionException | IllegalAccessException e) {
-            try { // to call fluent setter
-                PROPERTY_UTILS_BEAN.setProperty(object, field.getName(), value);
```

Comment:
I've tried the suggested option to flush caches of `java.beans.Introspector` and recreated the `PropertyUtilsBean` object, but it allows only to address the first issue (with overriden setter), but not the second (with overriden field)